### DISCUSSION TITLE
Songs play twice on generated playlists

### DISCUSSION THREAD

**Discussion by al090187** [music-assistant/support#1517] [Q&A]
Hi, when I select a song, and "start radio" it generates a playlist with songs listed once, but then plays each song twice. 

This doesn't happen when I play an album locally, or from Youtube Music integration. Only when I "start radio" where it generates the playlist itself.

Any suggestions?

**Comment by OzGav** [comment]
What provider are you using?

**Reply by al090187** [reply]
I assume YouTube Music. I have that, and a local network source enabled.

**Reply by OzGav** [reply]
@MarvinSchenkel ?

**Reply by al090187** [reply]
Yes, it is just using the youtube music integration. I just checked, and there is no "start radio" option on local files.

**Reply by MarvinSchenkel** [reply]
What device do you play the music on?

**Reply by al090187** [reply]
HA and MAss are on a Pi 4, and streaming to a Chromecast Audio. I will go check to see if it happens on my other google devices

**Reply by al090187** [reply]
OK, it did it on several of my chromecast audios, and has been since I installed it a week ago. If I enable or disable queue flow mode, it makes no difference

I just played it on my Google Home Mini, and it played fine....

Now it is playing fine on my Chromecast audio 😂 How odd.

**Reply by OzGav** [reply]
How is it going now? Has this seemingly resolved itself?

**Reply by al090187** [reply]
Hi, the problem does seem to have disappeared. How very odd. I will keep my ears open for it to replicate itself.

Just as I write this it repeats a song! Played 2 songs from the dynamic radio, then the 3rd it repeated! The two played appear in the "played" list, and the current song is still at the top of the next items.

The builtin log shows nothing for the time in question. It doesn't show a lot at all. Is there an option to have it log more to help diagnose it?

It just played that same song a third time. I checked that repeat is turned off. Could it be misreading the repeat flag somewhere?

**Reply by al090187** [reply]
OK, I skipped to the next track, it played OK, next track OK, then next track played 3 times, then it went to the next track, and played that three times, and then went to the next one. It's playing them 3 times now for some reason 🤷‍♂️

Here is a screenshot.
![image](https://github.com/music-assistant/hass-music-assistant/assets/18115688/af3f21d4-6412-4e7b-b87d-84aa87d7bdc9)

And a video of the player as it skips back to the start of the song.
https://github.com/music-assistant/hass-music-assistant/assets/18115688/dd86239f-1665-411f-bd99-c6148853e935

**Reply by MarvinSchenkel** [reply]
I've read reports from people that this is actually caused by chromecast rather than youtube music, since people seem to get this behavior when playing music on chromecast with "Enable flow mode" off for the player. Could you enable debug log on the chromecast provider and see if you can capture any good info from it when it happens next time?

**Reply by OzGav** [reply]
> Is there an option to have it log more to help diagnose it?

Yes go to MA SETTINGS>>PROVIDERS>>CHROMECAST>>CONFIGURE

**Comment by al090187** [comment]
OK, I have been trying all things. As you say, I think it is OK when flow mode is enabled. Although last night I thought it did it still when that was enabled, but I'll keep looking.
As for the log, is the below any help? the song (Tidal Wave - Sub Focus) jumped back to the start at 23:58:40.
Then after the 3rd play through, the song skipped to the next track (Chasing - All Stars), that was at 00:02:28. I missed a bit from this time, as it scrolled off the screen before I could copy it. I have included some logs from both times so you can compare. If you need any more, I can look tomorrow.

When it repeats it appears as though "Preloaded item 14" is sent at 23:58:26, which is good, because current item is listed as 13.
Then when it skips it says "Preloaded item 15" and current is 14. so it is somehow getting an increased position reference from somewhere. But, where is it getting these position references from? They don't correlate with the playlist positions as they were 6th and 7th. 

[playback log 11-9-23.txt](https://github.com/music-assistant/hass-music-assistant/files/12580997/playback.log.11-9-23.txt)

**Reply by MarvinSchenkel** [reply]
@marcelveldt any idea?

**Reply by marcelveldt** [reply]
There is some retry logic within Youtube Music when the auth fails, maybe that is somehow causing this issue ?
I have never heard about this issue with any of the other providers. Maybe just a simple wrong index issue ?

**Reply by MarvinSchenkel** [reply]
This is actually related to the discussion above. It seems to be related to the chromecast provider icw flow mode disabled. On Discord we got a report from someone who got this behaviour for Chromecast + Spotify.

**Reply by marcelveldt** [reply]
Wasn't that solved a while ago already ?

**Reply by OzGav** [reply]
I haven’t used radio much at all but I haven’t noticed it with Spotify+chromecast. I’ll have to try again and see if I experience this…?

**Reply by OzGav** [reply]
@marcelveldt  I just tried it with Spotify+Chromecast and yes I had repeating songs. I then tried with Spotify+Squeezelite and no problems. So @MarvinSchenkel might be correct that it is a Chromecast issue. Also tried local provider+Chromecast (no NOT radio mode) and first song repeated continuously.

I also had the problem that if seeked towards the end then 
(1) The track progress bar and counter would reset to zero even though the music would play from where I selected; and 
(2) the song would stop at the end of the track and pressing play again would result in the same track playing again...

**Comment by ctranter** [comment]
just adding my input here, i came to post the same thing. 
chromecast group,  tracks repeat randomly on local file playlists. switching to "flow mode" works.

**Reply by OzGav** [reply]
Can we have some followup on this issue. I am trying to reproduce on beta 74 without luck. If you can reproduce it then change the logging for the player to debug and capture everything over the period of duplicated playback.

**Reply by al090187** [reply]
Just got round to turning off Flow mode and trying it out, and I still had the problem whilst on b74. So I then enabled debug mode..... and it still repeated! I attach the whole file, but I noticed this....

22:39:00ish Repeated Barenaked Ladies - Falling for the first time
22:39:04:243 it "QUEUE_INSERT"ed Barber's Adagio for Strings (Ferry Corsten Remix) 
22:39:05.119 it "QUEUE_INSERT"ed Barber's Adagio for Strings (Ferry Corsten Remix) again
22:42:40ish it played Barber's Adagio for Strings (Ferry Corsten Remix)
No entry then for a "QUEUE_INSERT" for the next song, Eastern Jam - Chase + Status
22:49:16ish it repeated Barber's Adagio
then 22:49:18.084 a single "QUEUE_INSERT" for Eastern Jam
This then only played once, but a second after it started 2 "QUEUE_INSERT" commands appeared for the next song, and that played twice.

So sometimes it repeats the "queue insert" command on the first play, then as there is already a song in the queue, it doesn't do it next time and repeats the first song. Then as the queue is empty it adds the next song. No pattern to the repeats either though. 1, 1, 2, 3, 3, 4, 4, 5, 5, 6, 7, 7, 8, 9, 10, 10

Half way through item 9, Biffy Clyro - Bubbles, I pressed next track button, and it played Semi Charmed Life as expected, but also 2 QUEUE INSERTS for the next song, Return of the Mack. Pressing next again skipped to that song, so the second entry of Semi Charmed Life was ignored. So the problem doesn't manifest when manually skipping to the next track, only when playing does it repeat.

Anything else I can try, let me know.

[playback.log.02-11-23.txt](https://github.com/music-assistant/hass-music-assistant/files/13245535/playback.log.02-11-23.txt)

**Reply by OzGav** [reply]
Cheers. This will be looked at in due course. Thanks for your investigation and patience.
